models/ActivityModel.ts

```diff
@@ -5,13 +5,13 @@ import { UserModel } from './UserModel';
 
 @Entity('Activities')
 @Index('sliding_leaderboard_index', ['timestamp', 'pointsEarned'], { where: '"pointsEarned" > 0' })
+@Index('visible_activities_by_user_index', ['user', 'scope'], { where: 'scope = \'PUBLIC\' OR scope = \'PRIVATE\'' })
```

Comment:
I think this index works for both `public` and `public+private` queries but I'm not completely sure. Can you verify the index is being used for both queries? You can look up how to do that; IIRC it involves looking at the query planning using `EXPLAIN`.

Reply:
yep, `public`, `private`, and `public+private` all are indexed. 

It also looks like when selecting any column, not just user or scope, the index gets used. Is there any reason we need to specify `['user', 'scope']` for partial column indexes then?

Reply:
See the part about "order_nr" in the docs about partial indexes ([here](https://www.postgresql.org/docs/8.0/indexes-partial.html)). Without those columns, it'd still use the index but it'd scan the entire index looking for all rows matching a user instead of navigating to the part of the tree that has all rows for a user. Same for scope, if you wanted to get a user's public activities, you'd go to that part of the tree instead of scanning the entire index.